Modules/xASL_module_ASL.m

```diff
@@ -415,11 +420,22 @@
 		% Note: The paths x.P.(Pop_)Path_PWI4D_used can differ from a PWI4D obtained from a simple subtraction as we may have merged sessions,
 		% and the user or ExploreASL may have removed volumes from PWI4D. That's why we save and use x.P.(Pop_)Path_PWI4D_used in memory, such
 		% that the correct NIfTI will be used later in the pipeline (e.g., in xASL_wrp_VisualQC)
+
+		% Check again if we have multiple values for certain ASL parameters (PLD, LD, TE)
+		fprintf('%s\n','Sessions were concatenated. Checking again if LD, PLD, TE have multiple values...');
+		[tempPath, tempFile] = xASL_fileparts(x.P.Path_PWI4D_used);
+		tempASL = xASL_io_Nifti2Im(x.P.Path_PWI4D_used);
+		nVolumes = size(tempASL, 4);
+		x = xASL_module_ASL_MultiParameterParsing(x, fullfile(tempPath, [tempFile '.json']), nVolumes);
 	end
 end
+
 % Quantification is performed here according to ASL consensus paper (Alsop, MRM 2016)
 % Including PVC
 if ~x.mutex.HasState(StateName{iState}) && x.mutex.HasState(StateName{iState-4})
+	% Check multi-parametric quantification options
```

Comment:
See my answer to this above.
To repeat:
1. I moved them out of state-checks. 
2. multiParsing was run already at the start and should be only run again when merging was done.
3. QuantOptions should be run only once - cause it redefines options that are provided in dataPar.json and it would be difficult to run it twice and keep track of what was changed in ASL_module and what was provided at dataPar.json. So that's why QuantOptions is run separately from MultiParsing.